editor/src/core/model/element-template-utils.ts

```diff
@@ -543,92 +548,151 @@ export function insertJSXElementChild(
+                },
+                simpleValue,
+              )
+            } else {
+              if (isJSXFragment(clauseValue)) {
+                // Existing fragment, so add it in as appropriate.
+                let updatedChildren: Array<JSXElementChild>
+                if (indexPosition == null) {
+                  updatedChildren = clauseValue.children.concat(elementToInsert)
                 } else {
-                  // Something other than a fragment, so wrap that and the newly inserted element into a fragment.
-                  return jsxFragment(
-                    generateUidWithExistingComponents(projectContents),
-                    [clauseValue, elementToInsert],
-                    false,
+                  updatedChildren = Utils.addToArrayWithFill(
+                    elementToInsert,
+                    clauseValue.children,
+                    indexPosition,
+                    makeE,
                   )
                 }
+                return jsxFragment(clauseValue.uid, updatedChildren, clauseValue.longForm)
+              } else {
+                // Something other than a fragment, so wrap that and the newly inserted element into a fragment.
+                return jsxFragment(
+                  generateUidWithExistingComponents(projectContents),
+                  [clauseValue, elementToInsert],
+                  false,
+                )
               }
-            },
-            parentElement,
+            }
+          },
+          parentElement,
+        )
+      } else if (isJSXElementLike(parentElement)) {
+        let updatedChildren: Array<JSXElementChild>
+        if (indexPosition == null) {
+          updatedChildren = parentElement.children.concat(elementToInsert)
+        } else {
+          updatedChildren = Utils.addToArrayWithFill(
+            elementToInsert,
+            parentElement.children,
+            indexPosition,
+            makeE,
           )
-        } else if (isJSXElementLike(parentElement)) {
-          let updatedChildren: Array<JSXElementChild>
-          if (indexPosition == null) {
-            updatedChildren = parentElement.children.concat(elementToInsert)
-          } else {
-            updatedChildren = Utils.addToArrayWithFill(
-              elementToInsert,
-              parentElement.children,
-              indexPosition,
-              makeE,
+        }
+        return {
+          ...parentElement,
+          children: updatedChildren,
+        }
+      } else if (isJSXConditionalExpression(parentElement)) {
+        const trueCase = getConditionalCasePath(parentPath, 'true-case')
+        const parentMetadata = spyMetadata[EP.toString(parentPath)] ?? null
+        const isTrueCase =
+          getConditionalCase(trueCase, parentElement, parentMetadata, parentPath) === 'true-case'
+        let branch = isTrueCase ? { ...parentElement.whenTrue } : { ...parentElement.whenFalse }
+        switch (branch.type) {
```

Comment:
Rather than mutating `branch`, please can you extract this into a function

Reply:
https://github.com/concrete-utopia/utopia/pull/3453/commits/248e4598c3e068fc007ede6be6553839830571f0